### DISCUSSION TITLE
How to pass target specific flags when cross compiling?

### DISCUSSION THREAD

**Discussion by abique** [microsoft/vcpkg#41596] [Q&A] [answered]
Hi,

I wonder how to pass target specific flags when cross compiling?
Some programs have to be compiled for the machine currently building (host), like `tablegen.exe` in llvm, and some programs have to be built for the target like `ffmpeg.exe`.

How to specify different `CFLAGS` for the host and for the target?

Thanks!

**Comment by Osyotr** [comment, marked as answer]
Just set different values for `VCPKG_C_FLAGS` in host and target triplet.
https://learn.microsoft.com/en-us/vcpkg/users/triplets#vcpkg_cxx_flags

**Reply by abique** [reply]
This isn't my question.
When cross compiling, you have sometime to build intermediary programs that you have to execute during the build process. And you can't have ARM specific flags there if you're building on an intel machine while you still want to be able to have ARM specific flags for the target libraries.

**Reply by Osyotr** [reply]
> When cross compiling, you have sometime to build intermediary programs that you have to execute during the build process.

These are host dependencies and should be built with the host triplet.

**Reply by abique** [reply]
So you're saying that every ports that requires host tools during the build process needs to be built for the host as well?

It might be a tough job for libx11 or even llvm :thinking:

**Reply by Osyotr** [reply]
`llvm` probably doesn't need host tools, but `libx11` does:
https://github.com/microsoft/vcpkg/blob/6ab331448c76fbd1ca4fb599d374924e6d5e3024/ports/libx11/vcpkg.json#L10-L13

**Reply by dg0yt** [reply]
> It might be a tough job for libx11 or even llvm 🤔

It is a tough job, and it causes quite some breakage and limitations (qt5-base, icu, luajit, ...).

**Comment by dg0yt** [comment]
Target specific flags are actually triplet specific flags.  
Ports are only allowed to build for the target triplet.  
Host binaries must be built in the host triplet, and then be used from the target triplet via a dependency.  
This usually requires patching.
